### PR TITLE
SessionRecommender python api and example

### PR DESCRIPTION
**Pull request by tmacraft**

### CODE REVIEW

**Comment by songhappy** [comment]
separate session recommender python out, let this one go first
move play framework example into another PR

File: pyzoo/zoo/models/recommendation/session_recommender.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+if sys.version >= '3':
+    long = int
+    unicode = str
+
+
+class SessionRecommender(Recommender):
+
+    def __init__(self, item_count, item_embed, mlp_hidden_layers, rnn_hidden_layers,
+                 include_history, seq_length, his_length, bigdl_type="float"):
+        self.item_count = int(item_count)
```
Comment (songhappy) [inline pyzoo/zoo/models/recommendation/session_recommender.py]: make this order as the same as scala version

File: pyzoo/zoo/models/recommendation/session_recommender.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        """
+        jmodel = callBigDlFunc(bigdl_type, "loadSessionRecommender", path, weight_path)
+        model = KerasZooModel._do_load(jmodel, bigdl_type)
+        model.__class__ = SessionRecommender
+        return model
+
+
+
+
+
```
Comment (songhappy) [inline pyzoo/zoo/models/recommendation/session_recommender.py]: add test case as NCF

File: apps/recommendation-session-recommender/session_recommender.ipynb [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    "    return out"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "slide_udf = udf(lambda x: slide(x), \"array<array<double>>\")"
```
Comment (songhappy) [inline apps/recommendation-session-recommender/session_recommender.ipynb]: move utility functions to utils for more use cases

File: pyzoo/zoo/models/recommendation/session_recommender.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+     item_ount: The number of distinct items. Positive integer.
+     item_embed: The output size of embedding layer. Positive integer.
+     rnn_hidden_layers: Units of hidden layers for the mlp model. Array of positive integers.
+     session_length: The max number of items in the sequence of a session
+     include_history: Whether to include purchase history. Boolean. Default is true.
+     mlp_hidden_layers: Units of hidden layers for the mlp model. Array of positive integers.
+     history_length: The max number of items in the sequence of historical purchase
+     """
+    def __init__(self, item_count, item_embed, rnn_hidden_layers, session_length,
+                 include_history = False, mlp_hidden_layers = [10, 5], his_length = 2,
```
Comment (hkvision) [inline pyzoo/zoo/models/recommendation/session_recommender.py]: Change to `history_length`, same as line 40? `his` is a bit confusing...

Comment (hkvision) [inline pyzoo/zoo/models/recommendation/session_recommender.py]: Change to `history_length`, same as line 40? `his` is a bit confusing...

File: pyzoo/zoo/models/recommendation/session_recommender.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    # Arguments
+     item_ount: The number of distinct items. Positive integer.
+     item_embed: The output size of embedding layer. Positive integer.
+     rnn_hidden_layers: Units of hidden layers for the mlp model. Array of positive integers.
+     session_length: The max number of items in the sequence of a session
+     include_history: Whether to include purchase history. Boolean. Default is true.
+     mlp_hidden_layers: Units of hidden layers for the mlp model. Array of positive integers.
+     history_length: The max number of items in the sequence of historical purchase
+     """
+    def __init__(self, item_count, item_embed, rnn_hidden_layers, session_length,
```
Comment (hkvision) [inline pyzoo/zoo/models/recommendation/session_recommender.py]: Wondering do we have some suggested value for `rnn_hidden_layers` and `item_embed` as default?

**Comment by songhappy** [comment]
PR validation pass
http://10.239.47.211:18888/job/ZOO-PR-Validation/2513/

**Comment by songhappy** [comment]
@tmacraft please merge, and I will address the review comments in next PR of documentation and a fix of scala version

**Comment by hkvision** [comment]
@songhappy You can merge it for him....

**Comment by hkvision** [comment]
Where's the example?

**Pull request merged**